Mathlib/Data/Complex/Basic.lean

```diff
@@ -830,165 +925,196 @@ noncomputable def Complex.abs : AbsoluteValue ℂ ℝ
+  map_pow Complex.abs z n
 #align complex.abs_pow Complex.abs_pow
 
-@[simp]
-theorem abs_zpow (z : ℂ) (n : ℤ) : abs (z ^ n) = abs z ^ n :=
-  map_zpow₀ abs z n
+-- @[simp]
+/- Porting note: `simp` attribute removed as linter reports this can be proved
+by `simp only [@map_zpow₀]` -/
+theorem abs_zpow (z : ℂ) (n : ℤ) : Complex.abs (z ^ n) = Complex.abs z ^ n :=
+  map_zpow₀ Complex.abs z n
 #align complex.abs_zpow Complex.abs_zpow
 
-theorem abs_re_le_abs (z : ℂ) : |z.re| ≤ abs z :=
+theorem abs_re_le_abs (z : ℂ) : |z.re| ≤ Complex.abs z :=
   Real.abs_le_sqrt <| by
-    rw [norm_sq_apply, ← sq]
+    rw [normSq_apply, ← sq]
     exact le_add_of_nonneg_right (mul_self_nonneg _)
 #align complex.abs_re_le_abs Complex.abs_re_le_abs
 
-theorem abs_im_le_abs (z : ℂ) : |z.im| ≤ abs z :=
+theorem abs_im_le_abs (z : ℂ) : |z.im| ≤ Complex.abs z :=
   Real.abs_le_sqrt <| by
-    rw [norm_sq_apply, ← sq, ← sq]
+    rw [normSq_apply, ← sq, ← sq]
     exact le_add_of_nonneg_left (sq_nonneg _)
 #align complex.abs_im_le_abs Complex.abs_im_le_abs
 
-theorem re_le_abs (z : ℂ) : z.re ≤ abs z :=
+theorem re_le_abs (z : ℂ) : z.re ≤ Complex.abs z :=
   (abs_le.1 (abs_re_le_abs _)).2
 #align complex.re_le_abs Complex.re_le_abs
 
-theorem im_le_abs (z : ℂ) : z.im ≤ abs z :=
+theorem im_le_abs (z : ℂ) : z.im ≤ Complex.abs z :=
   (abs_le.1 (abs_im_le_abs _)).2
 #align complex.im_le_abs Complex.im_le_abs
 
 @[simp]
-theorem abs_re_lt_abs {z : ℂ} : |z.re| < abs z ↔ z.im ≠ 0 := by
-  rw [abs, AbsoluteValue.coe_mk, MulHom.coe_mk, Real.lt_sqrt (abs_nonneg _), norm_sq_apply,
+theorem abs_re_lt_abs {z : ℂ} : |z.re| < Complex.abs z ↔ z.im ≠ 0 := by
+  rw [Complex.abs, AbsoluteValue.coe_mk, MulHom.coe_mk, Real.lt_sqrt (abs_nonneg _), normSq_apply,
     _root_.sq_abs, ← sq, lt_add_iff_pos_right, mul_self_pos]
 #align complex.abs_re_lt_abs Complex.abs_re_lt_abs
 
 @[simp]
-theorem abs_im_lt_abs {z : ℂ} : |z.im| < abs z ↔ z.re ≠ 0 := by simpa using @abs_re_lt_abs (z * I)
+theorem abs_im_lt_abs {z : ℂ} : |z.im| < Complex.abs z ↔ z.re ≠ 0 := by
+  simpa using @abs_re_lt_abs (z * i)
 #align complex.abs_im_lt_abs Complex.abs_im_lt_abs
 
 @[simp]
-theorem abs_abs (z : ℂ) : |abs z| = abs z :=
-  abs_of_nonneg (abs.NonNeg _)
+theorem abs_abs (z : ℂ) : |Complex.abs z| = Complex.abs z :=
+  _root_.abs_of_nonneg (AbsoluteValue.nonneg _ z)
 #align complex.abs_abs Complex.abs_abs
 
-theorem abs_le_abs_re_add_abs_im (z : ℂ) : abs z ≤ |z.re| + |z.im| := by
-  simpa [re_add_im] using abs.add_le z.re (z.im * I)
+-- Porting note: probably should be golfed
+theorem abs_le_abs_re_add_abs_im (z : ℂ) : Complex.abs z ≤ |z.re| + |z.im| := by
+  repeat (rw [← abs_ofReal])
+  conv =>
+    lhs
+    rw [← Complex.re_add_im z]
+  simpa [re_add_im] using Complex.abs.add_le z.re (z.im * i)
 #align complex.abs_le_abs_re_add_abs_im Complex.abs_le_abs_re_add_abs_im
 
-theorem abs_le_sqrt_two_mul_max (z : ℂ) : abs z ≤ Real.sqrt 2 * max (|z.re|) (|z.im|) := by
+instance : NeZero (1 : ℝ) :=
+ ⟨by apply one_ne_zero⟩
```

Comment:
Please don't add new things like this without a porting note.

Reply:
Oops. Was meant to fix the next proof. Will add porting note.